### PR TITLE
[wasm] Remove the EnumCalendarInfo workaround from the pinvoke table …

### PR DESCRIPTION
**Pull request by vargaz**
…generator.

Its no longer needed because function pointers are now supported. Also, it wasn't used anymore because the pinvoke method was generated so its name changed.

### CODE REVIEW

File: src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs [hunk 1]
```diff
@@ -339,20 +339,13 @@ private static bool TryIsMethodGetParametersUnsupported(MethodInfo method, [NotN
     {
         var sb = new StringBuilder();
         var method = pinvoke.Method;
-        if (method.Name == "EnumCalendarInfo")
-        {
-            // FIXME: System.Reflection.MetadataLoadContext can't decode function pointer types
-            // https://github.com/dotnet/runtime/issues/43791
```
Comment (am11) [inline src/tasks/WasmAppBuilder/PInvokeTableGenerator.cs:345]: Same `FIXME` on line 316 can be deleted?

**Comment by vargaz** [comment]
It looks like the wasm build tests are using some old version of System.Reflection.MetadataLoadContext which still has the restriction. The error
message 'Parsing function pointer types in signatures is not supported' is no longer present in master of dotnet/runtime.

**Comment by lewing** [comment]
yeah, the task is pinning a version

**Comment by maraf** [comment]
> The error
> message 'Parsing function pointer types in signatures is not supported' is no longer present in master of dotnet/runtime.

I'm still seeing it on main. The `WasmAppBuilder` has a pinned version of `System.Reflection.MetadataLoadContext` to `6.0.1`.

**Comment by lambdageek** [comment]
I don't think we can remove the hack.  The issue is that all our MSBuild tasks are built for modern .NET but also for .NET Framework - when MSBuild is called from Visual Studio, the old .NET Framework versions are used which don't have the new APIs supported.  So you can `#ifdef` the hack code, but you can't remove it until VS uses a modern .NET for MSBuild (which won't be any time soon).

**Pull request closed without merging**